src/targets/gpu/hip.cpp

```diff
@@ -73,7 +81,7 @@ void* get_device_ptr(void* hptr)
     void* result = nullptr;
     auto status  = hipHostGetDevicePointer(&result, hptr, 0);
     if(status != hipSuccess)
-        MIGRAPHX_THROW("Failed getting device pointer: " + hip_error(status));
+        MIGRAPHX_THROW("Failed getting device pointer: " + hip_get_and_reset_error());
```

Comment:
This should still be `hip_error(status)`. We know what the error code is, so we should just use that instead of relying on the `hipLastError`.